### PR TITLE
Add Qwen3-Omni text-only config

### PR DESCRIPTION
**Pull request by hengtaoguo**
# Description

Add Qwen3-Omni text-only thinker model:
- `qwen3-omni-30b-a3b.yml` is adapted from `qwen3-30b-a3b.yml` MOE backbone, with updated `vocab_size=152064`. 
- Add params and hook mapping functions for `thinker.model`, by reusing Qwen3-MOE utils.

Next: Add different modalities components phase-by-phase.

# Tests

[Checkpoint](https://pantheon.corp.google.com/storage/browser/hengtaoguo-maxtext-logs/checkpoints/qwen3-omni-30b-a3b/unscanned/2025-10-30-02-00/0/items?pageState=(%22StorageObjectListTable%22:(%22f%22:%22%255B%255D%22))&e=13802955&mods=allow_workbench_image_override&project=tpu-prod-env-multipod) converted from [Qwen3-Omni-20B-A3B-Instruct](https://huggingface.co/Qwen/Qwen3-Omni-30B-A3B-Instruct) on a CPU VM. Omni's thinker is adapted from Qwen3-20B-A3B in MaxText:
```
python -m MaxText.utils.ckpt_conversion.to_maxtext MaxText/configs/base.yml hf_access_token=<hf_token> model_name=qwen3-omni-30b-a3b base_output_directory=gs://hengtaoguo-maxtext-logs/checkpoints/qwen3-omni-30b-a3b/unscanned/2025-10-30-02-00 scan_layers=false
```

Decode pure-text Qwen3-Omni on v5p-8:
```
python -m MaxText.decode MaxText/configs/base.yml model_name=qwen3-omni-30b-a3b tokenizer_type=huggingface tokenizer_path=src/MaxText/assets/qwen3-tokenizer load_parameters_path=gs://hengtaoguo-maxtext-logs/checkpoints/qwen3-omni-30b-a3b/unscanned/2025-10-30-02-00/0/items per_device_batch_size=1 run_name=ht_test max_prefill_predict_length=16 max_target_length=48 steps=1 async_checkpointing=false scan_layers=false use_multimodal=false prompt='Paris is the' hf_access_token=<hf_token> ici_data_parallelism=1 ici_expert_parallelism=8
```

Outputs:
```
Input `Paris is the` -> ` capital and most populous city of France, with an estimated population of 2,175,601 residents in 2023 within its administrative`
```

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: src/MaxText/utils/ckpt_conversion/utils/param_mapping.py [hunk 1]
```diff
@@ -814,6 +814,78 @@ def reshape_kernel(input_tensor, target_shape):
...
+  """
+  # Collect all modality mappings
+  mapping = {}
+
+  # Text mapping with "thinker." prefix, reusing QWEN3-MOE mapping function
+  num_experts_text = config["thinker_config"]["text_config"].get("num_experts", 0)
+  n_layers_text = config["thinker_config"]["text_config"]["num_hidden_layers"]
+  text_mapping = QWEN3_MAXTEXT_TO_HF_PARAM_MAPPING(
+      config={"num_hidden_layers": n_layers_text, "num_experts": num_experts_text}, scan_layers=scan_layers
+  )
```
[the remark this answers is not in the document]
Comment (hengtaoguo) [reply src/MaxText/utils/ckpt_conversion/utils/param_mapping.py:838]: Yeah I've been thinking about the design how to best reuse the Qwen3 MoE mappings. I don't want to (1) squeeze in complex if-else into the existing mappings (2) copy paste the entire ~200 lines of existing param/hook mappings. So I reused QWEN3_MAXTEXT_TO_HF_PARAM_MAPPING, then append `thinker.` before it. Would love to hear your suggestions! @aireenmei @parambole @RissyRan

File: src/MaxText/utils/ckpt_conversion/to_maxtext.py [hunk 2]
```diff
@@ -182,7 +194,7 @@ def main(argv: Sequence[str]) -> None:
   # Load HuggingFace model, config, and state_dict
   max_logging.log(f"Loading HuggingFace model: {model_id}...")
   hf_config_obj = AutoConfig.from_pretrained(model_id, token=hf_token)
-  hf_model = AutoModelForCausalLM.from_pretrained(model_id, token=hf_token)
+  hf_model = _get_hf_model(model_id, token=hf_token)
```
Comment (NicoGrande) [inline src/MaxText/utils/ckpt_conversion/to_maxtext.py:232]: Is this change necessary since there is no support for Qwen3OmniForCausalLM?

Comment (hengtaoguo) [reply src/MaxText/utils/ckpt_conversion/to_maxtext.py:232]: Thanks for pointing this out! Offline synced with Nico and for wider visibility, this is because Omni is a multimodal model and doesn't support calling from `AutoModelForCausalLM`. See Omni's official code demo: https://huggingface.co/Qwen/Qwen3-Omni-30B-A3B-Instruct#code-snippet

File: src/MaxText/utils/ckpt_conversion/to_maxtext.py [hunk 3]
```diff
@@ -182,6 +182,18 @@ def _build_single_axis_stacked_tensor(
   return np.stack(tensors_to_stack, axis=axis_to_stack)
 
 
+def _get_hf_model(model_id: str, token: str):
```
Comment (aireenmei) [inline src/MaxText/utils/ckpt_conversion/to_maxtext.py:185]: I like this design. I recall that Llama4 multimodal is also not supported in AutoModelForCausalLM and have a separate "Llama4ForConditionalGeneration" module. Have we supported Llama4 in this ckpt conversion script?

Comment (hengtaoguo) [reply src/MaxText/utils/ckpt_conversion/to_maxtext.py:185]: Thank you Aireen! Actually no, Llama4 is currently not supported in this tool, but there is an ongoing plan to migrate other models to here ([link](https://docs.google.com/document/d/1bPx0xvnjlChktgslC0kn7-9LQWihdjiqJNjZ0DfyVuI/edit?tab=t.0#heading=h.xgjl2srtytjt)).

**Review by aireenmei — APPROVED** [review]
